x-pack/plugins/ingest_manager/server/routes/agent/handlers.ts

```diff
@@ -315,6 +324,50 @@ export const putAgentsReassignHandler: RequestHandler<
   }
 };
 
+export const postBulkAgentsReassignHandler: RequestHandler<
+  undefined,
+  undefined,
+  TypeOf<typeof PostBulkAgentReassignRequestSchema.body>
+> = async (context, request, response) => {
+  if (!licenseService.isGoldPlus()) {
+    return response.customError({
+      statusCode: 500,
+      body: { message: 'Requires Gold license' },
+    });
+  }
+
+  const soClient = context.core.savedObjects.client;
+  try {
+    // Reassign by array of IDs
+    const result = Array.isArray(request.body.agents)
+      ? await AgentService.reassignAgents(
+          soClient,
+          { agentIds: request.body.agents },
+          request.body.policy_id
+        )
+      : await AgentService.reassignAgents(
+          soClient,
+          { kuery: request.body.agents },
+          request.body.policy_id
+        );
+    const body: PostBulkAgentReassignResponse = result.saved_objects.reduce((acc, so) => {
+      return {
+        ...acc,
+        [so.id]: {
+          success: !so.error,
+          error: so.error || undefined,
+        },
+      };
+    }, {});
+    return response.ok({ body });
+  } catch (e) {
```

Comment:
Mind replacing this with the new `defaultIngestErrorHandler` pattern?

Like https://github.com/elastic/kibana/pull/74409/commits/df19026250e4adf6214b0814d59d639bcf6e159f#diff-a4f9e2e344a4b58d7209d56878afed76

Reply:
nice, didn't know about this. I updated here as well as the equivalent single agent handler too for consistency, as well as the unenroll agent(s) handlers. I guess we have a task somewhere to replace and use it in all handlers?

Reply:
> I guess we have a task somewhere to replace and use it in all handlers?

@jen-huang better yet, a PR https://github.com/elastic/kibana/pull/77975 🎉